lib/most_frequent_integer.rb

```diff
@@ -3,5 +3,52 @@
 # exists, return nil.
 
 def find_most_frequent_integer(arr)
-  # put your code here
+
+sortarr = arr.sort
+indexarr = Array.new
+n = 0
+
+if sortarr.empty?
+    return nil
+else
+    i = sortarr[0]
+end
+
+while n < sortarr.length && i <= sortarr.last
+
+    indexarr[n] = sortarr.each_index.select{|x| sortarr[x] == i} 
+    i = i + 1
+    n = n + 1
+    
 end
+
+m = 0
+p = 0
+q = p + 1
+
+while m < indexarr.size && q < indexarr.size
+    if p == q
+        q = q + 1
+    else
+        
+        if indexarr[p].size > indexarr[q].size
+            q = q + 1
+            m = m + 1
+        else
+            
+            if indexarr[p].size == indexarr[q].size
+                return nil
+            else
+                
+                p = q
+                q = 0
+                m = m + 1
+            
+            end
+        end
+    end
+end
+
+return indexarr[p][1]
```

Comment:
This doesn't quite make sense to me.  `indexarr` is a collection of arrays, each representing the **indexes** where a given value occurs in the original array. So returning `indexarr[p][1]` is definitely returning a value that represents an index where the most-frequent integer occurred, and not the integer that is most-repeated. I think that this has been passing coincidentally because of the way our tests are written.